src/index.js

```diff
@@ -49,10 +50,23 @@ function splitGuide({
     return pify(fs.readFile)(file, 'utf8')
       .then(contents => ({file, contents}))
   }
+  
+  function promiseChunker(remainingArrayElements, chunkSize) {
```

Comment:
Actually, I know that sindre sorhus came up with a [bunch of promise related modules](https://github.com/sindresorhus/promise-fun) recently. Maybe he's got a utility like this...